lib/package-managers/npm.js

```diff
@@ -159,78 +205,92 @@ module.exports = {
             .then(versions => {
-                return _.last(pre ? versions : filterOutPrereleaseVersions(versions));
+                return _.last(options.pre ? versions : filterOutPrereleaseVersions(versions, options.pre));
             });
     },
 
     /**
      * @param {string} packageName
      * @param {string} currentVersion
-     * @param {boolean} pre
+     * @param {{}} options
      * @returns {Promise}
      */
-    greatest(packageName, currentVersion, pre) {
+    greatest(packageName, currentVersion, options) {
         return view(packageName, 'versions', currentVersion)
+            .then((versions) => {
+                return filterSatisfiesMinVersion(versions, options.enginesNode);
+            })
             .then(versions => {
-                return _.last(pre ? versions : filterOutPrereleaseVersions(versions));
+                return _.last(options.pre ? versions : filterOutPrereleaseVersions(versions, options.pre));
             });
     },
 
     /**
      * @param {string} packageName
      * @param {string} currentVersion
-     * @param {boolean} pre
+     * @param {{}} options
      * @returns {Promise}
      */
-    greatestMajor(packageName, currentVersion, pre) {
-        return view(packageName, 'versions', currentVersion).then(versions => {
-            const resultVersions = pre ? versions : filterOutPrereleaseVersions(versions);
-            return versionUtil.findGreatestByLevel(resultVersions, currentVersion, 'major');
-        });
+    greatestMajor(packageName, currentVersion, options) {
+        return view(packageName, 'versions', currentVersion)
+            .then((versions) => {
+                return filterSatisfiesMinVersion(versions, options.enginesNode);
+            })
+            .then(versions => {
+                const resultVersions = options.pre ? versions : filterOutPrereleaseVersions(versions, options.pre);
+                return versionUtil.findGreatestByLevel(resultVersions, currentVersion, 'major');
+            });
     },
 
     /**
      * @param {string} packageName
      * @param {string} currentVersion
-     * @param {boolean} pre
+     * @param {{}} options
      * @returns {Promise}
      */
-    greatestMinor(packageName, currentVersion, pre) {
-        return view(packageName, 'versions', currentVersion).then(versions => {
-            const resultVersions = pre ? versions : filterOutPrereleaseVersions(versions);
-            return versionUtil.findGreatestByLevel(resultVersions, currentVersion, 'minor');
-        });
+    greatestMinor(packageName, currentVersion, options) {
+        return view(packageName, 'versions', currentVersion)
+            .then((versions) => {
+                return filterSatisfiesMinVersion(versions, options.enginesNode);
+            })
+            .then(versions => {
+                const resultVersions = options.pre ? versions : filterOutPrereleaseVersions(versions, options.pre);
```

Comment:
Since `filterOutPrerelease versions` was modified to handle the `pre` check, I think you can simply this line as such:

```
const resultVersions = filterOutPrereleaseVersions(versions, options.pre)
```